mod-credits/src/main/java/gg/solarmc/loader/credits/CreditsKey.java

```diff
@@ -33,14 +33,14 @@
 import java.math.BigDecimal;
 import java.nio.file.Path;
 
-public class CreditsKey implements DataKey<Credits, CreditsManager> {
+public class CreditsKey implements DataKey<OnlineCredits, Credits, CreditsManager> {
 
-	CreditsKey(){}
+	CreditsKey() {}
 
 	public static final CreditsKey INSTANCE = new CreditsKey(); //specifying it as creditskey since it already implements the specification
 
 	@Override
-	public DataLoader<Credits> createLoader(CreditsManager dataManager, DataKeyInitializationContext context) {
+	public DataLoader<OnlineCredits, Credits> createLoader(CreditsManager dataManager, DataKeyInitializationContext context) {
```

Comment:
Youre gonna have to explain to me exactly what and what not offline implementations of the dataobject can implement, versus the online implementation.

Reply:
Offline data objects shouldn't have any sort of cached data. For example, it doesn't make sense to maintain the player's current balance in OfflineCredits.

Online data objects are free to maintain displayable/inaccurate data if they choose.

Reply:
got it! I had the general gist of it down but that clears things up.